h5c/vic/src/vic-webapp/src/app/shared/utils/vch/vch-utils.ts

```diff
@@ -414,12 +411,51 @@ export function processPayloadFromApiToUi (vch: VchApi): VchUi {
           volFileFolder: getDSNameAndFolder(vol.datastore).folderPath
         })
       }),
-
+    },
+    networks: {
+      bridgeNetwork: vch.network.bridge.port_group.id,
+      bridgeNetworkRange: vch.network.bridge.ip_range, // '172.16.0.0/12',
+      publicNetwork: vch.network.public.port_group.id,
+      publicNetworkIp: vch.network.public.gateway ? vch.network.public.gateway.routing_destinations.join(',') : '',
+      publicNetworkGateway: vch.network.public.gateway ? vch.network.public.gateway.address : '',
+      publicNetworkType: vch.network.public.nameservers && vch.network.public.nameservers.length > 0 ? 'static' : 'dhcp',
+      clientNetwork: vch.network.client.port_group.id,
+      clientNetworkIp: vch.network.client.gateway ? vch.network.client.gateway.routing_destinations.join(',') : '',
+      clientNetworkType: vch.network.client.nameservers && vch.network.client.nameservers.length > 0 ? 'static' : 'dhcp',
+      clientNetworkGateway: vch.network.client.gateway ? vch.network.client.gateway.address : '',
+      clientNetworkRouting: vch.network.client.nameservers.join(','),
+      managementNetwork: vch.network.management.port_group.id,
+      managementNetworkIp: vch.network.management.gateway ? vch.network.management.gateway.routing_destinations.join(',') : '',
+      managementNetworkType: vch.network.management.nameservers && vch.network.management.nameservers.length > 0 ? 'static' : 'dhcp',
+      managementNetworkGateway: vch.network.management.gateway ? vch.network.management.gateway.address : '',
+      managementNetworkRouting: vch.network.management.nameservers.join(','),
+      containerNetworks: vchNetworkContainerApiToUi(vch.network.container),
+      httpProxy: vch.network.httpProxy ? vch.network.httpProxy.split(':')[0] : '',
+      httpProxyPort: vch.network.httpProxy ? vch.network.httpProxy.split(':')[1] : '',
+      httpsProxy: vch.network.httpsProxy ? vch.network.httpsProxy.split(':')[0] : '',
+      httpsProxyPort: vch.network.httpsProxy ? vch.network.httpsProxy.split(':')[1] : '',
```

Comment:
this seems confusing because the payload and the interface seems to specify 'networks' as the object name rather than 'network'..so why is this vch.network rather than vch.networks?

Reply:
This is because they are two different interfaces:

"VchApi" is the interface for the vch payload retrieved from the API requests. I've made this interface based on the api documentation http://prntscr.com/jmb2jh

export interface VchApi {
  version?: string,
  name?: string,
  debug?: number,
  syslog_addr?: string
  compute?: VchApiCompute,
  network?: VchApiNetwork,
  storage?: VchApiStorage,
  auth?: VchApiAuth,
  endpoint?: VchApiEndpoint,
  registry?: VchApiRegistry,
  container?: VchApiContainer
  runtime?: VchApiRuntime,
}

"VchUi" is the interface to handle the vch information inside our UI. I've made this interface based on each of the creation wizard components that we already have on master, and also based on how they handle the vch information on their internal logic

https://github.com/vmware/vic-ui/blob/e8e3267cacb84b3e60770c43dbe0dc0dfc04341c/h5c/vic/src/vic-webapp/src/app/create-vch-wizard/storage-capacity/storage-capacity.component.ts#L177
https://github.com/vmware/vic-ui/blob/e8e3267cacb84b3e60770c43dbe0dc0dfc04341c/h5c/vic/src/vic-webapp/src/app/create-vch-wizard/storage-capacity/storage-capacity.component.ts#L177
https://github.com/vmware/vic-ui/blob/e8e3267cacb84b3e60770c43dbe0dc0dfc04341c/h5c/vic/src/vic-webapp/src/app/create-vch-wizard/networks/networks.component.ts#L339

export interface VchUi {
  general?: VchUiGeneral;
  computeCapacity?: VchUiCompute;
  storageCapacity?: VchUiStorage;
  networks?: VchUiNetwork;
  operations?: any, // work in progress
  registry?: any, // work in progress
  security?: any // work in progress
}

So basically we have two interfaces, and we have to convert the payloads between them:
1- VchApi -> is the api model and it has to be converted to VchUi after each vch GET request response.
2- VchUi -> is a more convenient structure to handle the information through our creation / configure flows and it has to be converted to VchApi before each vch POST / PUT / PATCH request.

In this particular case, we are converting the VchApi model -> "network" that comes from the API (http://prntscr.com/jmb2jh) into our internal VchUi model -> "networks" payload. The same happen for VchApi.compute -> VchUi.computeCapacity , VchApi.storage -> VchUi.storageCapacity , etc...

Reply:
Ok, it seems one is model and one is a viewModel ..we should think about the naming of these interfaces to be more intuitive or add some comment explaining what each one is for.